### PR TITLE
Sync: Only schedule initial full sync on JP install

### PR DESCRIPTION
**Pull request by lezama**
Alternative or complementary? to #5865

Tries to only start an initial Sync if we haven't done one before.

What do you think friends?

### CODE REVIEW

**Comment by lezama** [comment]
This should also improve the situation on #6041

File: tests/php/sync/test_class.jetpack-sync-integration.php [hunk 1]
```diff
@@ -37,16 +37,15 @@ function test_upgrading_sends_options_constants_and_callables() {
...
-		$initial_sync_without_users_config = array( 'options' => true, 'functions' => true, 'constants' => true, 'network_options' => true );
 		$initial_sync_with_users_config = array( 'options' => true, 'functions' => true, 'constants' => true, 'network_options' => true, 'users' => 'initial' );
 
 		do_action( 'updating_jetpack_version', '4.3', '4.2' );
 		$sync_status = Jetpack_Sync_Modules::get_module( 'full-sync' )->get_status();
 		$sync_config = $sync_status[ 'config' ];
 
-		$this->assertEquals( $initial_sync_without_users_config, $sync_config );
+		$this->assertNull( $sync_config );
 		$this->assertNotEquals( $initial_sync_with_users_config, $sync_config );
```
Comment (gravityrail) [inline tests/php/sync/test_class.jetpack-sync-integration.php]: This line is not necessary if we know the `$sync_config` is null

**Comment by gravityrail** [comment]
👍 good idea

**Comment by dereksmart** [comment]
4.5 merge 04ef80c

**Pull request merged**